torch/quantization/fx/quantization_patterns.py

```diff
@@ -1305,22 +1302,24 @@ def convert(self,
                     "call_method", "to", (op_out, torch.float16), {})
         else:
             assert is_reference
-            if dtypes in [(torch.quint8, torch.qint8, None)]:
-                load_arg(quantized=[0])(node.args)
+            # We can produce reference for a dtypes including
+            # (torch.quint8, torch.qint8, torch.qint32, torch.float16)
+            activation_post_process = \
+                self._maybe_get_last_node_only_observer(modules)
+            if activation_post_process is None:
```

Comment:
same as previous PR, could we check the dtype explicitly instead of using the implementation detail to check for it implicitly?